specs/commandline/verify.md

```diff
@@ -113,13 +124,46 @@ An example of output messages for a successful verification:
 Successfully verified signature for localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
 ```
 
+### Verify signatures on an OCI artifact with user metadata
+
+Use the `--user-metadata` flag to verify that provided key-value pairs are present in the payload of the valid signature.
+
+```shell
+# Verify signatures on the supplied OCI artifact identified by the digest and verify that io.wabbit-networks.data=foo is present in the signed payload
+notation verify localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9 --user-metadata io.wabbit-networks.data=foo
+```
+
+An example of output messages for a successful verification:
+
+```text
+Successfully verified signature for localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+
+User Metadata:
+- io.wabbit-networks.data : foo
+```
+
+### Verify signatures on an OCI artifact and format output as json
```

Comment:
I suggest creating a new PR for `--output` flag for both `notation sign` and `notation verify` commands. Besides successful cases, the follows examples are required as well
- example output of failure cases
- example output of successful or failure cases when additional `--debug` (debug logs) or `--verbose` flag (info logs) enabled

Reply:
The `--output` flag to specify the display format may be fine for `notation verify`. However, it may not be good for `notation sign`, especially for offline signing scenarios.

For example,
```
notation sign --oci net-monitor.tar@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9 --output net-monitor.sig
```

Reply:
can move this to the output PR when i create that one, but outputting the signature information as json is needed to allow consumption of the metadata programmatically, so for now, just proposing that we add it for `verify` only. is the concern about potential overlap with some other flag for sign?

Reply:
output reformatting moved to a separate PR, so resolving this